router/http/src/main/scala/io/buoyant/router/http/HeaderIdentifier.scala

```diff
@@ -0,0 +1,26 @@
+package io.buoyant.router.http
+
+import com.twitter.finagle.http.Request
+import com.twitter.finagle.{Path, Dtab}
+import com.twitter.finagle.buoyant.Dst
+import com.twitter.util.{Try, Future}
+import io.buoyant.router.RoutingFactory
+import io.buoyant.router.RoutingFactory.{UnidentifiedRequest, IdentifiedRequest, RequestIdentification}
+
+case class HeaderIdentifier(
+  prefix: Path,
+  header: String,
+  baseDtab: () => Dtab = () => Dtab.base
+) extends RoutingFactory.Identifier[Request] {
+
+  def apply(req: Request): Future[RequestIdentification[Request]] = {
+    req.headerMap.get(header) match {
+      case Some(value) =>
+        val path = Try(Path.read(value)).getOrElse(Path.Utf8(value))
```

Comment:
Hmmmm....  I suspect that this fallback logic will yield to some unexpected behavior.

```
scala> com.twitter.finagle.Path.Utf8("how/r/u").show
res0: String = /\x68\x6f\x77\x2f\x72\x2f\x75
```

It may be best to fail if it's not a parseable path.